src/python/WMCore/Services/DBS/DBSUtils.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+"""
+_DBSUtils_
+
+set of common utilities for DBS3Reader
+
+"""
+import json
+import urllib
+from urllib.parse import urlparse
+from urllib.parse import parse_qs
+from collections import defaultdict
+
+from Utils.CertTools import cert, ckey
+from dbs.apis.dbsClient import aggFileLumis, aggFileParents
+from WMCore.Services.pycurl_manager import getdata as multi_getdata
+
+
+def dbsListFileParents(dbsUrl, blocks, logger):
+    """
+    Concurrent counter part of DBS listFileParents API
+
+    :param dbsUrl: DBS URL
+    :param blocks: list of blocks
+    :param logger: logger
+    :return: list of file parents
+    """
+    urls = ['%s/fileparents?block_name=%s' % (dbsUrl, b.replace('#', '%23')) for b in blocks]
+    func = aggFileParents
+    uKey = 'block_name'
+    return getUrls(urls, func, uKey, logger)
+
+
+def dbsListFileLumis(dbsUrl, blocks, logger=None):
+    """
+    Concurrent counter part of DBS listFileLumis API
+
+    :param dbsUrl: DBS URL
+    :param blocks: list of blocks
+    :param logger: logger
+    :return: list of file lumis
+    """
+    urls = ['%s/filelumis?block_name=%s' % (dbsUrl, b.replace('#', '%23')) for b in blocks]
+    func = aggFileLumis
+    uKey = 'block_name'
+    return getUrls(urls, func, uKey, logger)
+
+
+def dbsBlockOrigin(dbsUrl, blocks, logger=None):
+    """
+    Concurrent counter part of DBS files API
+
+    :param dbsUrl: DBS URL
+    :param blocks: list of blocks
+    :param logger: logger
+    :return: list of block origins for a given parent lfns
+    """
+    urls = ['%s/blockorigin?block_name=%s' % (dbsUrl, b.replace('#', '%23')) for b in blocks]
+    func = None
+    uKey = 'block_name'
+    return getUrls(urls, func, uKey, logger)
+
+
+def dbsParentFilesGivenParentDataset(dbsUrl, parentDataset, fInfo, logger=None):
+    """
+    Obtain parent files for given fileInfo object
+
+    :param dbsUrl: DBS URL
+    :param parentDataset: parent dataset name
+    :param fInfo: file info object
+    :param logger: logger object
+    :return: list of parent files for given file info object
+    """
+    urls = []
+    for fileInfo in fInfo:
+        run = fileInfo['run_num']
+        lumis = urllib.parse.quote_plus(str(fileInfo['lumi_section_num']))
+        url = '%s/files?dataset=%s&run_num=%s&lumi_list=%s' \
+                % (dbsUrl, parentDataset, run, lumis)
+        if ':8443' not in url:
```

Comment:
I might be wrong, but I think this switch has already been covered by the portForward decorator, no?

Reply:
this code does not use any decorators, and no, I checked, the port was not changed, that's why I added the change.

Reply:
Then I'd suggest to use the exact same mechanism that Todor implemented in the last year, as you can see here:
https://github.com/dmwm/WMCore/pull/10330/files

in short:
```
portForwarder = PortForward(8443)
url = portForwarder(url)
```